Recharter proposal: 2018-2020

igrigorik opened the pull request:
This PR covers proposed updates to the 2018-2020 charter, as discussed in:
 - [proposal doc](https://docs.google.com/document/d/1Zx2orxWQhjN7fsIfrR2EHVhVzIanppKIyaqewrFJIFM/edit) 
 - [last week's WG call](https://lists.w3.org/Archives/Public/public-web-perf/2018Jun/0000.html)

Preview: https://rawgit.com/w3c/charter-webperf/18-20-charter/index.html

/cc @rniwa @past @toddreifsteck @slightlyoff

igrigorik committed:
extend new charter to June 2020

igrigorik committed:
define example focus areas within scope description

Per discusion on the call and in [1].

[1] https://docs.google.com/document/d/1Zx2orxWQhjN7fsIfrR2EHVhVzIanppKIyaqewrFJIFM/edit

igrigorik committed:
remove server-side profiling mention from out of scope

Server-Timing provides server-side profiling capabilities.

igrigorik committed:
define WICG coordination

igrigorik committed:
update repo and status links

- point directly to active github repos
- point to spec status spreadsheet

igrigorik committed:
remove memory pressure

No active development or implementer interest at this time.

igrigorik committed:
remove element timing

Under active development but there are no shipping implementations yet.
Removing from charter; we will continue incubatation in WICG.

igrigorik committed:
update spec draft state + completion

Added:
- device memory
- reporting
- network error logging

igrigorik committed:
update charter history with new changes

igrigorik committed:
note: Nav Timing L2+ will be merged into Resource Timing L2+

marcoscaceres commented on index.html:
```diff
@@ -204,66 +203,76 @@ <h3>
+              <p class="milestone"><b>Next Recommendation expected completion:</b> Q2 2019</p>
+
+              <p>Note: after Level 2 of Navigation Timing is ratified, future development of new Navigation Timing features will be done within Resource Timing; Navigation Timing will be merged with Resource Timing.</p>
             </dd>
```
👍

marcoscaceres reviewed (commented):
Great to see the incubations being moved back in.

plehegar commented:
We should add WHATWG in the external organization section.
Side note: once this get merged, I'll need to do a pass at the document to update the deliverables section with various reference drafts (to satisfy latest Patent Policy understanding).

plehegar commented on index.html:
```diff
@@ -366,7 +366,7 @@ <h3 id="w3c-coordination">W3C Groups</h3>
         <dt><a href="https://www.w3.org/community/wicg/">Web Platform Incubator Community Group</a></dt>
         <dd>
-          This group provides a lightweight venue for proposing, incubating and discussing new web platform features. Several proposals related to Web Performance were moved there: <a href="https://w3c.github.io/frame-timing/">Frame Timing</a>, <a href="https://w3c.github.io/server-timing/">Server Timing</a>, <a href="https://w3c.github.io/reporting/">Reporting</a>, and <a href="https://w3c.githu ...
+          This group provides a lightweight venue for proposing, incubating and discussing new web platform features. The Web Performance Working group will incubate and review new proposals that are within scope of our charter within the WICG. Once such WICG-incubated proposal has wide support and is available in at least one major browser, it may be adopted by the Web Performance Working group.
```
Did we conclude on one or two? Shouldn't we have a clear sign of interest from more than one before adopting a draft in the WG.

igrigorik replied:
So far we've said and enforced "one shipping" plus wider support from the group, which I personally interpret as at least one other vendor expressing interesting and support. That said, it may be worth spelling this out more explicitly.

> Once such WICG-incubated proposal is implemented and available in at least one major browser, and has support from one more, it may be adopted by the Web Performance Working group.

WDYT?

plehegar replied:
I like your rewording. Much better imho.

igrigorik committed:
add whatwg to external collab section

igrigorik commented:
@plehegar good catch on whatwg! Added.

plehegar commented:
One additional consideration: can we add a sentence or two related to the testing policy in section 1.2? eg. "ALL normative spec changes are generally expected to have a corresponding set of tests, either in the form of new tests or modifications to existing tests, or must include the rationale for why test updates are not required for the proposed update." (I left the WPT specific part out even if that's the current expectation).

igrigorik committed:
rework WICG adoption criteria

igrigorik committed:
add success criteria: tests for normative updates

igrigorik commented:
> "ALL normative spec changes are generally expected to have a corresponding set of tests, either in the form of new tests or modifications to existing tests, or must include the rationale for why test updates are not required for the proposed update."

👍 

Updated WICG adoption criteria and added above to success criteria.

rniwa commented:
The proposed charter looks good to us (Apple) although we're still interested in the memory pressure API.

igrigorik commented:
@rniwa great to hear, thanks for the feedback! 

Re, memory pressure: last time we did this exercise we added Memory Pressure into the normative deliverables section because we expected that we'd quickly converge on the spec and get 1+ shipping implementations out in quick order. For one reason or another that didn't happen.

On a tactical note, I removed memory pressure from deliverables to be consistent with our [incubation policy](https://github.com/w3c/charter-webperf/pull/37/files#diff-eacf331f0ffc35d4b482f1d15a887d3bL372). That doesn't mean we're abandoning that work. To the contrary, personally I'd love to see another push behind that effort. If we can drum up implementer interest, we can formally (re)adopt the spec into our charter — it's within the scope of our charter. 

As such, I don't think we need any changes to what we're proposing here, but just want to clarify and make sure we're all on the same page.

past commented:
This looks good for Mozilla as well, especially the NT/RT consolidation change Marcos noted above.

tdresser reviewed (approved):
Looks good to me!

yoavweiss reviewed (approved):
LGTM!

plehegar commented:
@toddreifsteck , ok to merge?

toddreifsteck commented:
@plehegar LGTM! @igrigorik Thanks for the work on driving this! Well done.
